build-system/server/new-server/transforms/utilities/script.ts

```diff
@@ -40,3 +40,12 @@ export function isValidScript(node: PostHTML.Node): node is ScriptNode {
   const src = new URL(attrs.src || '');
   return src.origin === VALID_CDN_ORIGIN && extname(src.pathname) === '.js';
 }
+
+export function isJsonScript(node: PostHTML.Node): boolean {
```

Comment:
Curious, what's the different between the return type here (`boolean`) and the return type of the similar `isValidScript` (`node is ScriptNode`)?

Reply:
so `node is ScriptNode` does type narrowing, wherein the compiler now now knows that node is not only a `PostHTML.Node` but is of the subtype `ScriptNode` which has additional attributes.

Reply:
possibly to make this more correct i should also make a type of JsonScript. i'll try and do that in https://github.com/ampproject/amphtml/pull/30292

Reply:
`node is ScriptNode` is a special type of return for TypeScript to tell the compiler an object is of a specific shape, and to trust the input value matches the shape based on success of the function.

Reply:
Fascinating, I've never encountered that syntax before. Thanks for explaining!